src/commands.jl

```diff
@@ -179,6 +181,25 @@ function next_line!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false
 end
 next_line!(frame::Frame, istoplevel::Bool=false) = next_line!(finish_and_return!, frame, istoplevel)
 
+"""
+    pc = until!(recurse, frame, line=nothing istoplevel=false)
+    pc = until!(frame, line=nothing, istoplevel=false)
+
+Execute until the current frame reaches a line greater than `line`. If `line == nothing`
+execute until the current frame reaches any line greater than the current line.
+"""
+function until_line!(@nospecialize(recurse), frame::Frame, line::Union{Nothing, Integer}=nothing, istoplevel::Bool=false)
+    pc = frame.pc
+    initialline, initialfile = linenumber(frame, pc), getfile(frame, pc)
+    line === nothing && (line = initialline + 1)
+    predicate(frame) = isexpr(pc_expr(frame), :return) || (linenumber(frame) >= line && getfile(frame) == initialfile)
+    pc = next_until!(predicate, frame, istoplevel)
+    (pc === nothing || isa(pc, BreakpointRef)) && return pc
+    maybe_step_through_kwprep!(recurse, frame, istoplevel)
+    maybe_next_call!(recurse, frame, istoplevel)
```

Comment:
Is there a risk this will bypass the chosen line? What if the chosen line has contents `x = 5`?

Reply:
Yes, it will, but that is not different from `next_line!`?

Reply:
OK, so it's a bug there too :smile:

Reply:
I guess the reason the code has originally liked to end up at calls is due to the many garbage assignments that happen, e.g. `t@_6 = y` etc.

If you prefer, I can remove the `maybe_next_call!` from here and `next_line!` and just remove the conditional https://github.com/JuliaDebug/Debugger.jl/blob/909afc889f7ed19679ab937cac64de6ed2b6a1e8/src/commands.jl#L48.

Reply:
What if we passed a predicate to `maybe_next_call!` looking for another increment in line number, and terminated advancement if it is satisfied?

That could be a separate PR and you could merge this as-is.

Reply:
So something like `advance_to_next_call_if_it_is_on_same_line_as_current_line` :P ? Perhaps that is a good compromise.

Reply:
Anyway, let's deal with the issues of exactly where we should stop in another issue / PR.